test/1_stdlib/Collection.swift

```diff
@@ -202,5 +202,41 @@ func testIsEmptyFirstLast() {
 }
 testIsEmptyFirstLast()
 
+/// A `CollectionType` that vends just the default implementations for
+/// `CollectionType` methods.
+struct CollectionOnly<T: CollectionType> : CollectionType {
+  var base: T
+
+  var startIndex: T.Index {
+    return base.startIndex
+  }
+
+  var endIndex: T.Index {
+    return base.endIndex
+  }
+
+  func generate() -> T.Generator {
+    return base.generate()
+  }
+
+  subscript(position: T.Index) -> T.Generator.Element {
+    return base[position]
+  }
+}
+
+func testFirstPerformance() {
+  // accessing `first` should not perform duplicate work on lazy collections
+  let col_ = (0..<10).lazy.filter({ print("filter: \($0)"); return (2..<8).contains($0) })
```

Comment:
It occurs to me that I'm not sure if `LoggingCollection` even does anything useful for me. That logs accesses to all of the `CollectionType` properties/methods. But I'm not particularly interested in logging how many times `startIndex` is accessed, what I really care about is how many times the filter predicate is evaluated on each element. Which comes full circle back to having a normal collection with a lazy filter and a side-effect-ful predicate.

So I think this basically means I just need to convert this over to `StdlibUnittest` but otherwise use the actual logic I already have.